R/provider-openai.R

```diff
@@ -278,24 +278,26 @@ method(as_json, list(ProviderOpenAI, Turn)) <- function(provider, x, ...) {
       list(role = "system", content = x@contents[[1]]@text)
     )
   } else if (x@role == "user") {
-    # Each tool result needs to go in its own message with role "tool"
-    is_tool <- map_lgl(x@contents, S7_inherits, ContentToolResult)
-    content <- as_json(provider, x@contents[!is_tool], ...)
-    if (length(content) > 0) {
+    data <- tool_results_separate_content(x)
```

Comment:
Do you need to update `chat_openai_responses()` too?

Reply:
Yes, I think so! I merged the changes but have been in meetings all day; I'll pick this back up on Monday and will take a look at `chat_openai_responses()` then too.